gateway.go

```diff
@@ -142,39 +143,44 @@ func (gw *Gateway) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 	}
 	log.Debugf("Computed response addresses %v", addrs)
 
-	m := new(dns.Msg)
-	m.SetReply(state.Req)
-
-	// If there's no match, fall through or return NXDOMAIN
+	// Fall through if no host matches
 	if len(addrs) == 0 {
 		if gw.Fall.Through(qname) {
 			return plugin.NextOrFailure(gw.Name(), gw.Next, ctx, w, r)
 		}
-
-		m.Rcode = dns.RcodeNameError
-		m.Ns = []dns.RR{gw.soa(state)}
-		if err := w.WriteMsg(m); err != nil {
-			log.Errorf("Failed to send a response: %s", err)
-		}
-		return 0, nil
```

Comment:
what happens if fallthrough is not configured but `len(addrs)` is 0?

Reply:
It will still return NXDOMAIN https://github.com/ori-edge/k8s_gateway/blob/df76f611f3ea464230cc617a2019e5f881340a09/gateway.go#L158-L161